rfc/rfc-37/rfc-37.md

```diff
@@ -0,0 +1,286 @@
+supports map like faster lookups by keys. Since, we will be storing stats/bloom for every file and the index will do
+lookups based on files, we should be able to benefit from the faster lookups in HFile. 
+
+<img src="metadata_index_1.png" alt="High Level Metadata Index Design" width="480"/>
+
+Following sections will talk about different partitions, key formats and then dive into the data and control flows.
+
+### MetaIndex/BloomFilter:
+
+A new partition `bloom_filter` will be added under the metadata table. Bloom filters from all the base files in the
+data table will be added here. Metadata table is already in the HFile format. The existing metadata payload schema will
+be extended and shared for this partition also. The type field will be used to detect the bloom filter payload record.
+Here is the schema for the bloom filter payload record.
+```
+           {
+            "doc": "Metadata about base file bloom filters",
+            "name": "BloomFilterMetadata",
+            "type": [
+                "null",
+                {
+                    "doc": "Base FileID and its BloomFilter details",
+                    "name": "HoodieMetadataBloomFilter",
+                    "type": "record",
+                    "fields": [
+                        {
+                            "doc": "Version/type of the bloom filter metadata",
+                            "name": "version",
+                            "type": "string"
+                        },
+                        {
+                            "doc": "Instant timestamp when this metadata was created/updated",
+                            "name": "timestamp",
+                            "type": "string"
+                        },
+                        {
+                            "doc": "Bloom filter binary byte array",
+                            "name": "bloomfilter",
+                            "type": "bytes"
+                        },
+                        {
+                            "doc": "True if this entry is valid",
+                            "name": "valid",
+                            "type": "boolean"
+                        }
+                    ]
+                }
+            ]
+        }
+```
+
+The key for the bloom filter record would be an encoded string representing the partition and base file combo. The
+partition and the file names are converted to deterministic hash based IDs, and then they are base64 encoded. Hash
+based IDs are easy to generate for the incoming new inserts records and for the lookup for the update records. 
+It and doesn't need any dictionary to be added for the reverse lookups. Hash bits are chosen based on the
+cardinality and the collision probability desired for the support max scale deployment. Base64 encoding the hash IDs
+further reduces the on-disk storage space for these keys.
+
+```
+key = base64_encode(hash64(partition name) + hash128(file path))
```

Comment:
We should shift before summing hashes to avoid commutation collisions

Reply:
Its actually string concat of base64 strings. fixed it.